napari/utils/event_handler.py

```diff
@@ -23,7 +26,7 @@ def register_listener(self, component):
             Object that contains callbacks for specific events. These are
             methods named according to an '_on_*_change' convention.
         """
-        self.components_to_update.append(component)
+        self.components.append(weakref.ref(component))
```

Comment:
Maybe add a comment here linking to Talley's [comment](https://github.com/napari/napari/pull/1391#issuecomment-653939143), so that future generations are not puzzled and/or tripped up by this?

Reply:
commend added